prog/dftb+/lib_io/indexselection.F90

```diff
@@ -502,4 +492,18 @@ function getTokenType_(token) result(tokenType)
   end function getTokenType_
 
 
+  !> Replacement for findloc(speciesNames, species, dim=1) to ensure GCC 7/8 compatibility
+  function findloc_(speciesNames, speciesName) result(iSpecies)
```

Comment:
At least make the function pure if you are [rearranging the deckchairs](https://en.wiktionary.org/wiki/rearrange_the_deck_chairs_on_the_Titanic) 
As soon as gcc7/8 pass out of common use, this should be ditched anyway.